nemoguardrails/embeddings/providers/__init__.py

```diff
@@ -87,7 +92,13 @@ def init_embedding_model(embedding_model: str, embedding_engine: str) -> Embeddi
     model_key = f"{embedding_engine}-{embedding_model}"
 
     if model_key not in _embedding_model_cache:
-        model = EmbeddingProviderRegistry().get(embedding_engine)(embedding_model)
+        embedding_engine = EmbeddingProviderRegistry().get(embedding_engine)
+
+        if "base_url" in inspect.signature(embedding_engine.__init__).parameters:
+            model = embedding_engine(embedding_model, base_url)
```

Comment:
It is safer to pass the `base_url` as a keyword argument, rather than a positional one. Maybe the assumption that it is the second positional parameter will break in the future.